src/libraries/System.Text.Json/tests/System.Text.Json.Tests/JsonNode/ParentPathRootTests.cs

```diff
@@ -180,5 +192,49 @@ public static void Parent_Object()
             parent.Add("MyProp", child);
             Assert.True(child.Options.Value.PropertyNameCaseInsensitive);
         }
+
+        public static IEnumerable<object[]> GetPath_ShouldReturnExpectedValue_TestData()
+        {
+            yield return new object[]
+            {
+                JsonNode.Parse("""{"$myRoot":{"foo['bar":"baz"}}""")["$myRoot"]["foo['bar"],
+                "$.$myRoot['foo[\\u0027bar']"
```

Comment:
If I'm reading these tests right, they're still not creating the correct paths.  `$myRoot` isn't valid for the dot syntax and needs to be bracketed.

Reply:
@Maximys would be possible to revisit this? Thanks!